src/mlpack/methods/reinforcement_learning/environment/pendulum.hpp

```diff
@@ -93,13 +93,17 @@ class Pendulum
    * @param maxAngularVelocity Maximum angular velocity.
    * @param maxTorque Maximum torque.
    * @param dt The differential value.
+   * @param angleThreshold The region about the upright position where the
+   *    state is considered terminal.
    */
   Pendulum(const double maxAngularVelocity = 8,
            const double maxTorque = 2.0,
-           const double dt = 0.05) :
+           const double dt = 0.05,
+           const double angleThreshold = M_PI/6) :
```

Comment:
Is that some default value you selected? Also there is a missing space between the ``/`` operator.

Reply:
Yes, is this too lenient? perhaps `M_PI/12` is better?

Reply:
I was mainly just wondering where this value comes from, but I agree ``M_PI/12`` might be better.

Reply:
Also, should I update HISTORY.MD? We didn't do it for PR #1931 so I'm not sure when we do this?

Reply:
Good point, I forgot about the HISTORY document, would be great to update the file as well.

Reply:
I wonder why do we need to impose constraint on pendulum using some angleThreshold, Isn't it better to just have terminals at `-PI `& `PI`, otherwise we won't ever be exploring with those states? Moreover this threshold still seems to be a bit large to me. Does it make sense?

Reply:
The idea here is that everything below the threshold would result in a terminal state. So we should choose something small, the issue I see here is that we don't terminate if we are above the threshold. So I guess it would make sense to introduce a maximal number of steps? I guess in this case we could also use the maximal number of steps as the terminal check?

Reply:
> the issue I see here is that we don't terminate if we are above the threshold

I'm not sure I understand, have I made a mistake in my terminal condition?

>  So I guess it would make sense to introduce a maximal number of steps?

I agree. I think if we do that for the Pendulum environment, we should do this for all environments.

Reply:
> I'm not sure I understand, have I made a mistake in my terminal condition?
No, there is notthing wrong there. As per my understanding Marcus wants to highlight the point that, as of now we don't terminate even when we reaches the terminal state, we just now have a function to lookup whether we arein terminal state or not. So, It could be a good to introduce maximum number of steps to act as a factor for deciding termination point.

> I agree. I think if we do that for the Pendulum environment, we should do this for all environments.
 Sounds good. If we all agree on this, than maybe the change can just be done within this PR.

Reply:
> > the issue I see here is that we don't terminate if we are above the threshold
> 
> I'm not sure I understand, have I made a mistake in my terminal condition?

No issue with the. terminal condition, just an issue at which point we stop, but that applies for all tasks.

>I agree. I think if we do that for the Pendulum environment, we should do this for all environments.
> > Sounds good. If we all agree on this, than maybe the change can just be done within this PR.

Fine with me, but I think the default should either be no limit or a pretty high number. What do you think?

Reply:
I think we could make the default value 0, in which case we won't make a check for termination.

Reply:
Do you think we should maybe add a `Log::Info` stating the reason for termination of the episode, since now there are two possibilities?

Reply:
Good idea.